deploy/role.yaml

```diff
@@ -10,6 +10,7 @@ rules:
   - pods
   verbs:
   - get
+  - create
```

Comment:
I'd suggest to move the `secrets` entry to the next one, so that the operator is not granted permission to create pods.